src/coreclr/jit/lclmorph.cpp

```diff
@@ -677,6 +677,34 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 PopValue();
                 break;
 
+            case GT_SUB:
+            {
+                Value& rhs = TopValue(0);
+                Value& lhs = TopValue(1);
+                if (m_compiler->opts.OptimizationEnabled() && lhs.IsAddress() && rhs.IsAddress() &&
+                    (lhs.LclNum() == rhs.LclNum()) && (rhs.Offset() <= lhs.Offset()) &&
+                    FitsIn<int>(lhs.Offset() - rhs.Offset()))
+                {
+                    // TODO-Correctness: Due to inlining we may end up with incorrectly typed SUB trees here.
```

Comment:
```suggestion
                    // TODO-Bug: Due to inlining we may end up with incorrectly typed SUB trees here.
```
Since that's in `conventions.txt`...